distributed/node.py

```diff
@@ -77,24 +77,32 @@ def stop_services(self):
     def service_ports(self):
         return {k: v.port for k, v in self.services.items()}
 
-    def _setup_logging(self, logger):
-        self._deque_handler = DequeHandler(
-            n=dask.config.get("distributed.admin.log-length")
-        )
-        self._deque_handler.setFormatter(
-            logging.Formatter(dask.config.get("distributed.admin.log-format"))
-        )
-        logger.addHandler(self._deque_handler)
-        weakref.finalize(self, logger.removeHandler, self._deque_handler)
+    def _setup_logging(self, logger, log_file):
```

Comment:
I haven't looked closely at the new implementation (sorry) but is it possible to do something like

```python
if log_file:
    handler = logging.FileHandler(log_file)
else:
    handler = DequeHandler(...)
```

And leave the rest as is? (though your changes to `get_logs` would still be needed).

Reply:
Sorry, ignore my comment, I see that that's essentially what you're doing.

Though I don't quite see how stuff makes it *to* log-file, but maybe that's handled elsewhere.

Reply:
> Though I don't quite see how stuff makes it to log-file, but maybe that's handled elsewhere.

Yeah my assumption here is that the supervising process (cloud-init, systemd, etc) will capture additional output that logging doesn't (like tracebacks) and also log its own stuff around process lifecycle.